fuel-vm/src/interpreter/memory.rs

```diff
@@ -287,6 +327,75 @@ where
     }
 }
 
+pub(crate) fn push_selected_registers(
+    memory: &mut [u8; MEM_SIZE],
+    mut sp: RegMut<SP>,
+    hp: Reg<HP>,
+    pc: RegMut<PC>,
+    program_regs: &ProgramRegisters,
+    segment: ProgramRegistersSegment,
+    bitmask: Imm24,
+) -> Result<(), RuntimeError> {
+    let bitmask = bitmask.to_u32();
+
+    // First compute the new stack pointer, as that's the only error condition
+    let count = bitmask.count_ones();
+    let stack_range = MemoryRange::new(*sp, (count as u64) * 8)?;
+    let new_sp = stack_range.words().end;
+    if new_sp > *hp {
```

Comment:
If sp == hp then we have a collision. Maybe we could refactor some of the code used in `stack_pointer_overflow` to be reusable here?

```suggestion
    if new_sp >= *hp {
```

Reply:
Fixed in 0e00edf6952dcfa2e779b1178e762c64d2f895ee. The current code style is poorly composable, so the result is a bit ugly, but at least they share the validation logic now.